source/uri-options/tests/connection-options.yml

```diff
@@ -0,0 +1,87 @@
+tests:
+    -
+        description: "Valid connection and timeout options are parsed correctly"
+        uri: "mongodb://example.com/?appname=URI-OPTIONS-SPEC-TEST&connectTimeoutMS=20000&heartbeatFrequencyMS=5000&localThresholdMS=3000&maxIdleTimeMS=50000&replicaSet=uri-options-spec&retryWrites=true&serverSelectionTimeoutMS=15000&socketTimeoutMS=7500"
+        valid: true
+        warning: false
+        hosts: ~
```

Comment:
I also do not understand the logic behind specifying a null host here. If the idea is to reuse connection string spec runner with the expectation that that runner parses the host, then the host here should be "example.com", and this test should fail as written. If the idea is to ignore hosts when running URI options tests then they should not be specified in the spec tests.